dispatch/queue.h

```diff
@@ -1346,6 +1346,18 @@ dispatch_assert_queue_not(dispatch_queue_t queue)
 #define dispatch_assert_queue_not_debug(q) dispatch_assert_queue_not(q)
 #endif
 
+
+#ifdef __ANDROID__
+/*!
+ * @handler dispatch_thread_detach_callback
+ *
+ * Hook to be able to detach threads before they exit from the Java JVM.
+ * If JNI has been used on a thread on Android it needs to have been
+ * "detached" before the thread exits or the application will crash.
+ */
+DISPATCH_EXPORT
+void (*_dispatch_thread_detach_callback)(void);
```

Comment:
for the second time this is **NOT** appropriate for an API header, this belongs to an internal one.
if the swift overlay is not built seeing the internal headers, then either try to fix that, or extern the prototype here again. it is an internal implementation detail between the overlay and libdispatch that should never be exposed anywhere.

this **HAS** to move to queue_internal.h, or maybe "private/private.h" this could be appropriate as well, and would likely be seen by the overlay.